WebDevStudios/wds-blocks
# Feature/block variations

Closes #107 

### Link
N/A

### Description
- Add starter block variations:
  - CTA Cover 
     - Cover block variation
     - H1 Heading
     - Optional Paragraph block
     - Button block - outline variation and set to 75% width
     - Cover image placeholder
     - Blocks is set to `cover` but the `alignfull` class does not work on the `wd_s` theme. Screenshot below shows block full width on the `twenty twenty-one theme`
  
  - Services 
     - Columns block variation
     - With 3 columns
     - Innerblocks:
        - Image
        - Heading
        - Paragraph

- No front end styling added, will be done on theme instead. 
- Added custom icon support for block icons, instead of just using `dashicons` - this is optional but a nice to have since we sometimes have a lot of blocks and dashicons won't be able to cover all of them.  

### Screenshot
![Block-Variations-–-wd_s](https://user-images.githubusercontent.com/1948204/123666966-445dd600-d86c-11eb-8d38-791758707cac.png)


https://user-images.githubusercontent.com/1948204/123666983-47f15d00-d86c-11eb-8585-2e85e7684527.mp4


### Steps to verify the solution
1. Create/edit a page
2. Add the `CTA Cover` block variation and add content.
3. Add the `Services` block variation and add content.
4. Check to make sure they are both showing.

### Other

-   [ ] Is this issue accessible? (Section 508/WCAG 2.1AA)
-   [ ] Does this pass CrossBrowserTesting.com? (Edge, Safari, Chrome, Firefox)
-   [ ] Will this pull request require [updating the wiki?](https://github.com/WebDevStudios/wds-block-starter/wiki)
